src/hooks/useSessionContext.ts

```diff
@@ -0,0 +1,12 @@
+import { useContext } from "react";
+import { SessionContext } from "../context/SessionContext";
+
+const useSessionContext = () => {
+    const sessionContext = useContext(SessionContext);
+    if (sessionContext === undefined) {
+        throw new Error("Context is undefined");
```

Comment:
does it give the error anytime? you have the context defined, then why will it be undefined? Also if is undefined you can write the below code

```suggestion
    if (!sessionContext) {
        throw new Error("Context is undefined");
```